docs/sphinx/hpx-functions.rst

```diff
@@ -0,0 +1,102 @@
+.. _HPX Functions:
+
+=============
+HPX Functions
+=============
+
+This document describes the prototypes, the allowable arguments of the listed
+HPX functions.
+
+async
+=====
+
+Async in C++ allows running asynchronous programming. Async programming is useful
+in compute heavy implementations. Basic async function would look like
+
+.. code-block:: text
+   #include <future>
+   using namespace hpx;
+   int main() {
+    future<int> f = async([]() {});
```

Comment:
This should be either:
```
future<void> f = async([]() {});
```
or
```
future<int> f = async([]() -> int { return 42; });
```
as the future returned from async holds the exact same type as is returned by the function.

Reply:
8c51a2c.